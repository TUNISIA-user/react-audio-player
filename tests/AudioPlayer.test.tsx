import React from "react";
import { expect, test, vi } from "vitest";
import renderer from "react-test-renderer"
import userEvent from "@testing-library/user-event";
import { render, act, screen } from "@testing-library/react";
import { AudioPlayer } from "../src/components/AudioPlayer";

describe("Test AudioPlayer", () => {
  test("AudioPlayer snapshot full", () => {
    const audioControls = renderer.create(
      <AudioPlayer 
        src="https://actions.google.com/sounds/v1/alarms/digital_watch_alarm_long.ogg"
        width={350}
        trackHeight={75}
      />
    )
    const tree = audioControls.toJSON();
    expect(tree).toMatchSnapshot();
  })

  test("AudioPlayer snapshot full - no visualize", () => {
    const audioControls = renderer.create(
      <AudioPlayer 
        src="https://actions.google.com/sounds/v1/alarms/digital_watch_alarm_long.ogg"
        width={350}
        trackHeight={75}
        visualise={false}
      />
    )
    const tree = audioControls.toJSON();
    expect(tree).toMatchSnapshot();
  })

  test("AudioPlayer snapshot full - no skip", () => {
    const audioControls = renderer.create(
      <AudioPlayer 
        src="https://actions.google.com/sounds/v1/alarms/digital_watch_alarm_long.ogg"
        width={350}
        trackHeight={75}
        allowSkip={false}
      />
    )
    const tree = audioControls.toJSON();
    expect(tree).toMatchSnapshot();
  })

  test("AudioPlayer snapshot full - no loop", () => {
    const audioControls = renderer.create(
      <AudioPlayer 
        src="https://actions.google.com/sounds/v1/alarms/digital_watch_alarm_long.ogg"
        width={350}
        trackHeight={75}
        showLoopOption={false}
      />
    )
    const tree = audioControls.toJSON();
    expect(tree).toMatchSnapshot();
  })

  test("AudioPlayer snapshot full - no volume control", () => {
    const audioControls = renderer.create(
      <AudioPlayer 
        src="https://actions.google.com/sounds/v1/alarms/digital_watch_alarm_long.ogg"
        width={350}
        trackHeight={75}
        showVolumeControl={false}
      />
    )
    const tree = audioControls.toJSON();
    expect(tree).toMatchSnapshot();
  })

  test("AudioPlayer snapshot full - no seekbar", () => {
    const audioControls = renderer.create(
      <AudioPlayer 
        src="https://actions.google.com/sounds/v1/alarms/digital_watch_alarm_long.ogg"
        width={350}
        trackHeight={75}
        hideSeekBar={false}
      />
    )
    const tree = audioControls.toJSON();
    expect(tree).toMatchSnapshot();
  })

  test("AudioPlayer snapshot full - minimal", () => {
    const audioControls = renderer.create(
      <AudioPlayer 
        src="https://actions.google.com/sounds/v1/alarms/digital_watch_alarm_long.ogg"
        width={350}
        trackHeight={75}
        minimal={true}
      />
    )
    const tree = audioControls.toJSON();
    expect(tree).toMatchSnapshot();
  })
})
