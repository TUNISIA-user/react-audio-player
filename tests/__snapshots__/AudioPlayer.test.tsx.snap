// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test AudioPlayer > AudioPlayer snapshot full - minimal 1`] = `
<div
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#EFEFEF",
      "borderRadius": "5px",
      "display": "flex",
      "flexDirection": "column",
      "gap": 10,
      "height": "max-content",
      "padding": "7px 10px",
      "position": "relative",
      "width": 350,
    }
  }
>
  <div
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "gap": 10,
        "justifyContent": "space-evenly",
        "width": 350,
      }
    }
  >
    <div
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "gap": 15,
          "gridColumn": "2 / 2",
        }
      }
    >
      <img
        data-testid="play-pause"
        onClick={[Function]}
        src="/src/icons/play.svg"
        style={
          {
            "cursor": "pointer",
            "height": "20px",
            "width": "20px",
          }
        }
        title="Play"
      />
    </div>
    <div
      style={
        {
          "display": "grid",
          "gridTemplateColumns": "1fr",
          "gridTemplateRows": "1fr",
          "height": 75,
          "width": 244.99999999999997,
        }
      }
    >
      <div
        data-testid="seek-trackbar-trackbar"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        style={
          {
            "alignItems": "center",
            "cursor": "pointer",
            "display": "flex",
            "gridColumn": 1,
            "gridRow": 1,
            "height": "15px",
            "placeSelf": "center",
            "width": "100%",
          }
        }
        tabIndex={0}
      >
        <div
          style={
            {
              "alignItems": "center",
              "backgroundColor": "rgba(140, 140, 140)",
              "cursor": "pointer",
              "display": "flex",
              "height": 1,
              "position": "relative",
              "width": "100%",
            }
          }
        >
          <div
            style={
              {
                "backgroundColor": "rgba(140, 140, 140)",
                "borderRadius": "50%",
                "cursor": "pointer",
                "height": "10px",
                "left": "NaN%",
                "marginLeft": -5,
                "opacity": 1,
                "position": "absolute",
                "transition": "opacity 0.75s ease-in-out",
                "width": "10px",
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      data-testid="timer"
      style={
        {
          "color": "#000000",
          "fontSize": 14,
        }
      }
    >
      00:00
    </div>
  </div>
</div>
`;

exports[`Test AudioPlayer > AudioPlayer snapshot full - no loop 1`] = `
<div
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#EFEFEF",
      "borderRadius": "5px",
      "display": "flex",
      "flexDirection": "column",
      "gap": 10,
      "height": "max-content",
      "padding": "10px 20px 15px",
      "position": "relative",
      "width": 350,
    }
  }
>
  <div
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "gap": 15,
        "justifyContent": "space-evenly",
        "width": 350,
      }
    }
  >
    <div
      data-testid="timer"
      style={
        {
          "color": "#000000",
          "fontSize": 14,
        }
      }
    >
      00:00
    </div>
    <div
      style={
        {
          "display": "grid",
          "gridTemplateColumns": "1fr",
          "gridTemplateRows": "1fr",
          "height": 75,
          "width": 244.99999999999997,
        }
      }
    >
      <div
        data-testid="seek-trackbar-trackbar"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        style={
          {
            "alignItems": "center",
            "cursor": "pointer",
            "display": "flex",
            "gridColumn": 1,
            "gridRow": 1,
            "height": "15px",
            "placeSelf": "center",
            "width": "100%",
          }
        }
        tabIndex={0}
      >
        <div
          style={
            {
              "alignItems": "center",
              "backgroundColor": "rgba(140, 140, 140)",
              "cursor": "pointer",
              "display": "flex",
              "height": 1,
              "position": "relative",
              "width": "100%",
            }
          }
        >
          <div
            style={
              {
                "backgroundColor": "rgba(140, 140, 140)",
                "borderRadius": "50%",
                "cursor": "pointer",
                "height": "10px",
                "left": "NaN%",
                "marginLeft": -5,
                "opacity": 1,
                "position": "absolute",
                "transition": "opacity 0.75s ease-in-out",
                "width": "10px",
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      data-testid="timer"
      style={
        {
          "color": "#000000",
          "fontSize": 14,
        }
      }
    >
      00:00
    </div>
  </div>
  <div
    style={
      {
        "display": "grid",
        "gridTemplateColumns": "1fr 1fr 1fr",
        "placeItems": "center",
        "width": "98%",
      }
    }
  >
    <div
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "gap": 15,
          "gridColumn": "2 / 2",
        }
      }
    >
      <img
        data-testid="skip-back"
        onClick={[Function]}
        src="/src/icons/skip.svg"
        style={
          {
            "cursor": "pointer",
            "height": 16,
            "transform": "rotate(180deg)",
            "width": 16,
          }
        }
        title="Skip backwards"
      />
      <img
        data-testid="play-pause"
        onClick={[Function]}
        src="/src/icons/play.svg"
        style={
          {
            "cursor": "pointer",
            "height": "20px",
            "width": "20px",
          }
        }
        title="Play"
      />
      <img
        data-testid="skip-forward"
        onClick={[Function]}
        src="/src/icons/skip.svg"
        style={
          {
            "cursor": "pointer",
            "height": 16,
            "width": 16,
          }
        }
        title="Skip forwards"
      />
    </div>
    <div
      style={
        {
          "display": "flex",
          "gap": 8,
          "justifySelf": "flex-end",
          "width": 80,
        }
      }
    >
      <img
        onClick={[Function]}
        src="/src/icons/volume.svg"
        style={
          {
            "cursor": "pointer",
            "height": 16,
            "width": 16,
          }
        }
        title="Mute"
      />
      <div
        data-testid="volume-trackbar"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        style={
          {
            "alignItems": "center",
            "cursor": "pointer",
            "display": "flex",
            "gridColumn": 1,
            "gridRow": 1,
            "height": "15px",
            "placeSelf": "center",
            "width": "100%",
          }
        }
        tabIndex={0}
      >
        <div
          style={
            {
              "alignItems": "center",
              "backgroundColor": "rgba(140, 140, 140)",
              "cursor": "pointer",
              "display": "flex",
              "height": 1,
              "position": "relative",
              "width": "100%",
            }
          }
        >
          <div
            style={
              {
                "backgroundColor": "rgba(140, 140, 140)",
                "borderRadius": "50%",
                "cursor": "pointer",
                "height": "10px",
                "left": "75%",
                "marginLeft": -5,
                "opacity": 1,
                "position": "absolute",
                "transition": "opacity 0.75s ease-in-out",
                "width": "10px",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test AudioPlayer > AudioPlayer snapshot full - no seekbar 1`] = `
<div
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#EFEFEF",
      "borderRadius": "5px",
      "display": "flex",
      "flexDirection": "column",
      "gap": 10,
      "height": "max-content",
      "padding": "10px 20px 15px",
      "position": "relative",
      "width": 350,
    }
  }
>
  <div
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "gap": 15,
        "justifyContent": "space-evenly",
        "width": 350,
      }
    }
  >
    <div
      data-testid="timer"
      style={
        {
          "color": "#000000",
          "fontSize": 14,
        }
      }
    >
      00:00
    </div>
    <div
      style={
        {
          "display": "grid",
          "gridTemplateColumns": "1fr",
          "gridTemplateRows": "1fr",
          "height": 75,
          "width": 244.99999999999997,
        }
      }
    >
      <div
        data-testid="seek-trackbar-trackbar"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        style={
          {
            "alignItems": "center",
            "cursor": "pointer",
            "display": "flex",
            "gridColumn": 1,
            "gridRow": 1,
            "height": "15px",
            "placeSelf": "center",
            "width": "100%",
          }
        }
        tabIndex={0}
      >
        <div
          style={
            {
              "alignItems": "center",
              "backgroundColor": "rgba(140, 140, 140)",
              "cursor": "pointer",
              "display": "flex",
              "height": 1,
              "position": "relative",
              "width": "100%",
            }
          }
        >
          <div
            style={
              {
                "backgroundColor": "rgba(140, 140, 140)",
                "borderRadius": "50%",
                "cursor": "pointer",
                "height": "10px",
                "left": "NaN%",
                "marginLeft": -5,
                "opacity": 1,
                "position": "absolute",
                "transition": "opacity 0.75s ease-in-out",
                "width": "10px",
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      data-testid="timer"
      style={
        {
          "color": "#000000",
          "fontSize": 14,
        }
      }
    >
      00:00
    </div>
  </div>
  <div
    style={
      {
        "display": "grid",
        "gridTemplateColumns": "1fr 1fr 1fr",
        "placeItems": "center",
        "width": "98%",
      }
    }
  >
    <img
      onClick={[Function]}
      src="/src/icons/repeat-one.svg"
      style={
        {
          "cursor": "pointer",
          "height": 16,
          "justifySelf": "flex-start",
          "width": 16,
        }
      }
      title="Loop"
    />
    <div
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "gap": 15,
          "gridColumn": "2 / 2",
        }
      }
    >
      <img
        data-testid="skip-back"
        onClick={[Function]}
        src="/src/icons/skip.svg"
        style={
          {
            "cursor": "pointer",
            "height": 16,
            "transform": "rotate(180deg)",
            "width": 16,
          }
        }
        title="Skip backwards"
      />
      <img
        data-testid="play-pause"
        onClick={[Function]}
        src="/src/icons/play.svg"
        style={
          {
            "cursor": "pointer",
            "height": "20px",
            "width": "20px",
          }
        }
        title="Play"
      />
      <img
        data-testid="skip-forward"
        onClick={[Function]}
        src="/src/icons/skip.svg"
        style={
          {
            "cursor": "pointer",
            "height": 16,
            "width": 16,
          }
        }
        title="Skip forwards"
      />
    </div>
    <div
      style={
        {
          "display": "flex",
          "gap": 8,
          "justifySelf": "flex-end",
          "width": 80,
        }
      }
    >
      <img
        onClick={[Function]}
        src="/src/icons/volume.svg"
        style={
          {
            "cursor": "pointer",
            "height": 16,
            "width": 16,
          }
        }
        title="Mute"
      />
      <div
        data-testid="volume-trackbar"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        style={
          {
            "alignItems": "center",
            "cursor": "pointer",
            "display": "flex",
            "gridColumn": 1,
            "gridRow": 1,
            "height": "15px",
            "placeSelf": "center",
            "width": "100%",
          }
        }
        tabIndex={0}
      >
        <div
          style={
            {
              "alignItems": "center",
              "backgroundColor": "rgba(140, 140, 140)",
              "cursor": "pointer",
              "display": "flex",
              "height": 1,
              "position": "relative",
              "width": "100%",
            }
          }
        >
          <div
            style={
              {
                "backgroundColor": "rgba(140, 140, 140)",
                "borderRadius": "50%",
                "cursor": "pointer",
                "height": "10px",
                "left": "75%",
                "marginLeft": -5,
                "opacity": 1,
                "position": "absolute",
                "transition": "opacity 0.75s ease-in-out",
                "width": "10px",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test AudioPlayer > AudioPlayer snapshot full - no skip 1`] = `
<div
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#EFEFEF",
      "borderRadius": "5px",
      "display": "flex",
      "flexDirection": "column",
      "gap": 10,
      "height": "max-content",
      "padding": "10px 20px 15px",
      "position": "relative",
      "width": 350,
    }
  }
>
  <div
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "gap": 15,
        "justifyContent": "space-evenly",
        "width": 350,
      }
    }
  >
    <div
      data-testid="timer"
      style={
        {
          "color": "#000000",
          "fontSize": 14,
        }
      }
    >
      00:00
    </div>
    <div
      style={
        {
          "display": "grid",
          "gridTemplateColumns": "1fr",
          "gridTemplateRows": "1fr",
          "height": 75,
          "width": 244.99999999999997,
        }
      }
    >
      <div
        data-testid="seek-trackbar-trackbar"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        style={
          {
            "alignItems": "center",
            "cursor": "pointer",
            "display": "flex",
            "gridColumn": 1,
            "gridRow": 1,
            "height": "15px",
            "placeSelf": "center",
            "width": "100%",
          }
        }
        tabIndex={0}
      >
        <div
          style={
            {
              "alignItems": "center",
              "backgroundColor": "rgba(140, 140, 140)",
              "cursor": "pointer",
              "display": "flex",
              "height": 1,
              "position": "relative",
              "width": "100%",
            }
          }
        >
          <div
            style={
              {
                "backgroundColor": "rgba(140, 140, 140)",
                "borderRadius": "50%",
                "cursor": "pointer",
                "height": "10px",
                "left": "NaN%",
                "marginLeft": -5,
                "opacity": 1,
                "position": "absolute",
                "transition": "opacity 0.75s ease-in-out",
                "width": "10px",
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      data-testid="timer"
      style={
        {
          "color": "#000000",
          "fontSize": 14,
        }
      }
    >
      00:00
    </div>
  </div>
  <div
    style={
      {
        "display": "grid",
        "gridTemplateColumns": "1fr 1fr 1fr",
        "placeItems": "center",
        "width": "98%",
      }
    }
  >
    <img
      onClick={[Function]}
      src="/src/icons/repeat-one.svg"
      style={
        {
          "cursor": "pointer",
          "height": 16,
          "justifySelf": "flex-start",
          "width": 16,
        }
      }
      title="Loop"
    />
    <div
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "gap": 15,
          "gridColumn": "2 / 2",
        }
      }
    >
      <img
        data-testid="play-pause"
        onClick={[Function]}
        src="/src/icons/play.svg"
        style={
          {
            "cursor": "pointer",
            "height": "20px",
            "width": "20px",
          }
        }
        title="Play"
      />
    </div>
    <div
      style={
        {
          "display": "flex",
          "gap": 8,
          "justifySelf": "flex-end",
          "width": 80,
        }
      }
    >
      <img
        onClick={[Function]}
        src="/src/icons/volume.svg"
        style={
          {
            "cursor": "pointer",
            "height": 16,
            "width": 16,
          }
        }
        title="Mute"
      />
      <div
        data-testid="volume-trackbar"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        style={
          {
            "alignItems": "center",
            "cursor": "pointer",
            "display": "flex",
            "gridColumn": 1,
            "gridRow": 1,
            "height": "15px",
            "placeSelf": "center",
            "width": "100%",
          }
        }
        tabIndex={0}
      >
        <div
          style={
            {
              "alignItems": "center",
              "backgroundColor": "rgba(140, 140, 140)",
              "cursor": "pointer",
              "display": "flex",
              "height": 1,
              "position": "relative",
              "width": "100%",
            }
          }
        >
          <div
            style={
              {
                "backgroundColor": "rgba(140, 140, 140)",
                "borderRadius": "50%",
                "cursor": "pointer",
                "height": "10px",
                "left": "75%",
                "marginLeft": -5,
                "opacity": 1,
                "position": "absolute",
                "transition": "opacity 0.75s ease-in-out",
                "width": "10px",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test AudioPlayer > AudioPlayer snapshot full - no visualize 1`] = `
<div
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#EFEFEF",
      "borderRadius": "5px",
      "display": "flex",
      "flexDirection": "column",
      "gap": 10,
      "height": "max-content",
      "padding": "10px 20px 15px",
      "position": "relative",
      "width": 350,
    }
  }
>
  <div
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "gap": 15,
        "justifyContent": "space-evenly",
        "width": 350,
      }
    }
  >
    <div
      data-testid="timer"
      style={
        {
          "color": "#000000",
          "fontSize": 14,
        }
      }
    >
      00:00
    </div>
    <div
      style={
        {
          "display": "grid",
          "gridTemplateColumns": "1fr",
          "gridTemplateRows": "1fr",
          "height": 75,
          "width": 244.99999999999997,
        }
      }
    >
      <div
        data-testid="seek-trackbar-trackbar"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        style={
          {
            "alignItems": "center",
            "cursor": "pointer",
            "display": "flex",
            "gridColumn": 1,
            "gridRow": 1,
            "height": "15px",
            "placeSelf": "center",
            "width": "100%",
          }
        }
        tabIndex={0}
      >
        <div
          style={
            {
              "alignItems": "center",
              "backgroundColor": "rgba(140, 140, 140)",
              "cursor": "pointer",
              "display": "flex",
              "height": 1,
              "position": "relative",
              "width": "100%",
            }
          }
        >
          <div
            style={
              {
                "backgroundColor": "rgba(140, 140, 140)",
                "borderRadius": "50%",
                "cursor": "pointer",
                "height": "10px",
                "left": "NaN%",
                "marginLeft": -5,
                "opacity": 1,
                "position": "absolute",
                "transition": "opacity 0.75s ease-in-out",
                "width": "10px",
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      data-testid="timer"
      style={
        {
          "color": "#000000",
          "fontSize": 14,
        }
      }
    >
      00:00
    </div>
  </div>
  <div
    style={
      {
        "display": "grid",
        "gridTemplateColumns": "1fr 1fr 1fr",
        "placeItems": "center",
        "width": "98%",
      }
    }
  >
    <img
      onClick={[Function]}
      src="/src/icons/repeat-one.svg"
      style={
        {
          "cursor": "pointer",
          "height": 16,
          "justifySelf": "flex-start",
          "width": 16,
        }
      }
      title="Loop"
    />
    <div
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "gap": 15,
          "gridColumn": "2 / 2",
        }
      }
    >
      <img
        data-testid="skip-back"
        onClick={[Function]}
        src="/src/icons/skip.svg"
        style={
          {
            "cursor": "pointer",
            "height": 16,
            "transform": "rotate(180deg)",
            "width": 16,
          }
        }
        title="Skip backwards"
      />
      <img
        data-testid="play-pause"
        onClick={[Function]}
        src="/src/icons/play.svg"
        style={
          {
            "cursor": "pointer",
            "height": "20px",
            "width": "20px",
          }
        }
        title="Play"
      />
      <img
        data-testid="skip-forward"
        onClick={[Function]}
        src="/src/icons/skip.svg"
        style={
          {
            "cursor": "pointer",
            "height": 16,
            "width": 16,
          }
        }
        title="Skip forwards"
      />
    </div>
    <div
      style={
        {
          "display": "flex",
          "gap": 8,
          "justifySelf": "flex-end",
          "width": 80,
        }
      }
    >
      <img
        onClick={[Function]}
        src="/src/icons/volume.svg"
        style={
          {
            "cursor": "pointer",
            "height": 16,
            "width": 16,
          }
        }
        title="Mute"
      />
      <div
        data-testid="volume-trackbar"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        style={
          {
            "alignItems": "center",
            "cursor": "pointer",
            "display": "flex",
            "gridColumn": 1,
            "gridRow": 1,
            "height": "15px",
            "placeSelf": "center",
            "width": "100%",
          }
        }
        tabIndex={0}
      >
        <div
          style={
            {
              "alignItems": "center",
              "backgroundColor": "rgba(140, 140, 140)",
              "cursor": "pointer",
              "display": "flex",
              "height": 1,
              "position": "relative",
              "width": "100%",
            }
          }
        >
          <div
            style={
              {
                "backgroundColor": "rgba(140, 140, 140)",
                "borderRadius": "50%",
                "cursor": "pointer",
                "height": "10px",
                "left": "75%",
                "marginLeft": -5,
                "opacity": 1,
                "position": "absolute",
                "transition": "opacity 0.75s ease-in-out",
                "width": "10px",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test AudioPlayer > AudioPlayer snapshot full - no volume control 1`] = `
<div
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#EFEFEF",
      "borderRadius": "5px",
      "display": "flex",
      "flexDirection": "column",
      "gap": 10,
      "height": "max-content",
      "padding": "10px 20px 15px",
      "position": "relative",
      "width": 350,
    }
  }
>
  <div
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "gap": 15,
        "justifyContent": "space-evenly",
        "width": 350,
      }
    }
  >
    <div
      data-testid="timer"
      style={
        {
          "color": "#000000",
          "fontSize": 14,
        }
      }
    >
      00:00
    </div>
    <div
      style={
        {
          "display": "grid",
          "gridTemplateColumns": "1fr",
          "gridTemplateRows": "1fr",
          "height": 75,
          "width": 244.99999999999997,
        }
      }
    >
      <div
        data-testid="seek-trackbar-trackbar"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        style={
          {
            "alignItems": "center",
            "cursor": "pointer",
            "display": "flex",
            "gridColumn": 1,
            "gridRow": 1,
            "height": "15px",
            "placeSelf": "center",
            "width": "100%",
          }
        }
        tabIndex={0}
      >
        <div
          style={
            {
              "alignItems": "center",
              "backgroundColor": "rgba(140, 140, 140)",
              "cursor": "pointer",
              "display": "flex",
              "height": 1,
              "position": "relative",
              "width": "100%",
            }
          }
        >
          <div
            style={
              {
                "backgroundColor": "rgba(140, 140, 140)",
                "borderRadius": "50%",
                "cursor": "pointer",
                "height": "10px",
                "left": "NaN%",
                "marginLeft": -5,
                "opacity": 1,
                "position": "absolute",
                "transition": "opacity 0.75s ease-in-out",
                "width": "10px",
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      data-testid="timer"
      style={
        {
          "color": "#000000",
          "fontSize": 14,
        }
      }
    >
      00:00
    </div>
  </div>
  <div
    style={
      {
        "display": "grid",
        "gridTemplateColumns": "1fr 1fr 1fr",
        "placeItems": "center",
        "width": "98%",
      }
    }
  >
    <img
      onClick={[Function]}
      src="/src/icons/repeat-one.svg"
      style={
        {
          "cursor": "pointer",
          "height": 16,
          "justifySelf": "flex-start",
          "width": 16,
        }
      }
      title="Loop"
    />
    <div
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "gap": 15,
          "gridColumn": "2 / 2",
        }
      }
    >
      <img
        data-testid="skip-back"
        onClick={[Function]}
        src="/src/icons/skip.svg"
        style={
          {
            "cursor": "pointer",
            "height": 16,
            "transform": "rotate(180deg)",
            "width": 16,
          }
        }
        title="Skip backwards"
      />
      <img
        data-testid="play-pause"
        onClick={[Function]}
        src="/src/icons/play.svg"
        style={
          {
            "cursor": "pointer",
            "height": "20px",
            "width": "20px",
          }
        }
        title="Play"
      />
      <img
        data-testid="skip-forward"
        onClick={[Function]}
        src="/src/icons/skip.svg"
        style={
          {
            "cursor": "pointer",
            "height": 16,
            "width": 16,
          }
        }
        title="Skip forwards"
      />
    </div>
  </div>
</div>
`;

exports[`Test AudioPlayer > AudioPlayer snapshot full 1`] = `
<div
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#EFEFEF",
      "borderRadius": "5px",
      "display": "flex",
      "flexDirection": "column",
      "gap": 10,
      "height": "max-content",
      "padding": "10px 20px 15px",
      "position": "relative",
      "width": 350,
    }
  }
>
  <div
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "gap": 15,
        "justifyContent": "space-evenly",
        "width": 350,
      }
    }
  >
    <div
      data-testid="timer"
      style={
        {
          "color": "#000000",
          "fontSize": 14,
        }
      }
    >
      00:00
    </div>
    <div
      style={
        {
          "display": "grid",
          "gridTemplateColumns": "1fr",
          "gridTemplateRows": "1fr",
          "height": 75,
          "width": 244.99999999999997,
        }
      }
    >
      <div
        data-testid="seek-trackbar-trackbar"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        style={
          {
            "alignItems": "center",
            "cursor": "pointer",
            "display": "flex",
            "gridColumn": 1,
            "gridRow": 1,
            "height": "15px",
            "placeSelf": "center",
            "width": "100%",
          }
        }
        tabIndex={0}
      >
        <div
          style={
            {
              "alignItems": "center",
              "backgroundColor": "rgba(140, 140, 140)",
              "cursor": "pointer",
              "display": "flex",
              "height": 1,
              "position": "relative",
              "width": "100%",
            }
          }
        >
          <div
            style={
              {
                "backgroundColor": "rgba(140, 140, 140)",
                "borderRadius": "50%",
                "cursor": "pointer",
                "height": "10px",
                "left": "NaN%",
                "marginLeft": -5,
                "opacity": 1,
                "position": "absolute",
                "transition": "opacity 0.75s ease-in-out",
                "width": "10px",
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      data-testid="timer"
      style={
        {
          "color": "#000000",
          "fontSize": 14,
        }
      }
    >
      00:00
    </div>
  </div>
  <div
    style={
      {
        "display": "grid",
        "gridTemplateColumns": "1fr 1fr 1fr",
        "placeItems": "center",
        "width": "98%",
      }
    }
  >
    <img
      onClick={[Function]}
      src="/src/icons/repeat-one.svg"
      style={
        {
          "cursor": "pointer",
          "height": 16,
          "justifySelf": "flex-start",
          "width": 16,
        }
      }
      title="Loop"
    />
    <div
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "gap": 15,
          "gridColumn": "2 / 2",
        }
      }
    >
      <img
        data-testid="skip-back"
        onClick={[Function]}
        src="/src/icons/skip.svg"
        style={
          {
            "cursor": "pointer",
            "height": 16,
            "transform": "rotate(180deg)",
            "width": 16,
          }
        }
        title="Skip backwards"
      />
      <img
        data-testid="play-pause"
        onClick={[Function]}
        src="/src/icons/play.svg"
        style={
          {
            "cursor": "pointer",
            "height": "20px",
            "width": "20px",
          }
        }
        title="Play"
      />
      <img
        data-testid="skip-forward"
        onClick={[Function]}
        src="/src/icons/skip.svg"
        style={
          {
            "cursor": "pointer",
            "height": 16,
            "width": 16,
          }
        }
        title="Skip forwards"
      />
    </div>
    <div
      style={
        {
          "display": "flex",
          "gap": 8,
          "justifySelf": "flex-end",
          "width": 80,
        }
      }
    >
      <img
        onClick={[Function]}
        src="/src/icons/volume.svg"
        style={
          {
            "cursor": "pointer",
            "height": 16,
            "width": 16,
          }
        }
        title="Mute"
      />
      <div
        data-testid="volume-trackbar"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        style={
          {
            "alignItems": "center",
            "cursor": "pointer",
            "display": "flex",
            "gridColumn": 1,
            "gridRow": 1,
            "height": "15px",
            "placeSelf": "center",
            "width": "100%",
          }
        }
        tabIndex={0}
      >
        <div
          style={
            {
              "alignItems": "center",
              "backgroundColor": "rgba(140, 140, 140)",
              "cursor": "pointer",
              "display": "flex",
              "height": 1,
              "position": "relative",
              "width": "100%",
            }
          }
        >
          <div
            style={
              {
                "backgroundColor": "rgba(140, 140, 140)",
                "borderRadius": "50%",
                "cursor": "pointer",
                "height": "10px",
                "left": "75%",
                "marginLeft": -5,
                "opacity": 1,
                "position": "absolute",
                "transition": "opacity 0.75s ease-in-out",
                "width": "10px",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
